cli/flox/doc/flox-activate.md

```diff
@@ -155,18 +155,15 @@ See [`manifest.toml(5)`](./manifest.toml.md) for more details on shell hooks.
 
 ## Variables used by `flox activate`
 
-`$FLOX_SHELL`
-:  When launching an interactive sub-shell, Flox launches the shell specified in
-   `$FLOX_SHELL` if it is set.
-   When printing a shell for sourcing in the current shell,
-   Flox will produce a script suitable for `$FLOX_SHELL` if it is set.
-
-`$SHELL`
-:  When launching an interactive sub-shell, Flox launches the shell specified in
-   `$SHELL` if it is set and `$FLOX_SHELL` is not set.
-   When printing a shell for sourcing in the current shell,
-   Flox will produce a script suitable for `$SHELL` if it is set
-   and `$FLOX_SHELL` is not set and Flox can't detect the parent shell.
+`$FLOX_SHELL`, `$FLOX_SHELL_NOCLOBBER`, `$SHELL`
+:  When launching an interactive sub-shell, Flox invokes the shell specified in
+   `$FLOX_SHELL_NOCLOBBER` if set, then `$FLOX_SHELL` if that is set,
+   and finally falls back to invoke `$SHELL` by default.
+   Conversely, when performing an "in place" activation
+   Flox will attempt to detect its parent shell type
+   unless overridden by the `$FLOX_SHELL_NOCLOBBER` variable,
+   and if it cannot detect its parent shell type then will
+   produce a script with syntax determined by `$SHELL`.
```

Comment:
I don't think this really explains what NOCLOBBER is and for now I'd rather just leave the docs as is

Reply:
The previous explanation didn't explain what FLOX_SHELL was, it explained how it worked, so I followed suit with my rendition. I agree it would make sense to explain the motivation behind FLOX_SHELL as the means of using flox activate to launch other shells - perhaps adding a sentence to explain that at the beginning would be an improvement?

Reply:
Updated doc following conversations with @mkenigs ... re-requesting review and resolving conversation.